src/groupeddataframe/grouping.jl

```diff
@@ -443,71 +655,145 @@ colwise(fns::Union{AbstractVector, Tuple}, d::AbstractDataFrame) = [f(d[i]) for
 colwise(f, gd::GroupedDataFrame) = [colwise(f, g) for g in gd]
 
 """
-Split-apply-combine in one step; apply `f` to each grouping in `d`
-based on columns `col`
-
-```julia
-by(d::AbstractDataFrame, cols, f::Function; sort::Bool = false)
-by(f::Function, d::AbstractDataFrame, cols; sort::Bool = false)
-```
-
-### Arguments
-
-* `d` : an AbstractDataFrame
-* `cols` : a column indicator (Symbol, Int, Vector{Symbol}, etc.)
-* `f` : a function to be applied to groups; expects each argument to
-  be an AbstractDataFrame
-* `sort`: sort row groups (no sorting by default)
-
-### Details
+    by(d::AbstractDataFrame, keys, cols => f...; sort::Bool = false)
+    by(d::AbstractDataFrame, keys; (colname = cols => f)..., sort::Bool = false)
+    by(d::AbstractDataFrame, keys, f; sort::Bool = false)
+    by(f, d::AbstractDataFrame, keys; sort::Bool = false)
```

Comment:
same here - `f` can be tuple or named tuple I think. Can it be a `Pair` (same question to `combine`)